lib/grpc/GrpcService.ts

```diff
@@ -431,6 +432,32 @@ class GrpcService {
     }
   }
 
+  /**
+   * See [[Service.listTrades]]
+   */
+  public listTrades: grpc.handleUnaryCall<xudrpc.ListTradesRequest, xudrpc.ListTradesResponse> = async (call, callback) => {
+    try {
+      const listTrades = await this.service.listTrades(call.request.toObject());
+      const response = new xudrpc.ListTradesResponse();
+      const trades: xudrpc.Trade[] = [];
+      listTrades.forEach(async (trade: TradeInstance) => {
+        const grpcTrade = new xudrpc.Trade();
+        const order = grpcTrade.getMakerOrder();
```

Comment:
This should be `const makerOrder = await trade.getMakerOrder();` - otherwise we are just getting the empty `Order` message from the grpc response we are trying to construct.

I put together this method you can use to convert an `OrderInstance` into a grpc order:

```
      const getGrpcOrderFromOrderInstance = (order: OrderInstance) => {
        const grpcOrder = new xudrpc.Order();
        grpcOrder.setCreatedAt(order.createdAt);
        grpcOrder.setId(order.id);
        grpcOrder.setIsOwnOrder(order.nodeId === undefined)
        if (order.localId) {
          grpcOrder.setLocalId(order.localId);
        }
        grpcOrder.setPairId(order.pairId);
        // TODO: set peer pub key if order.nodeId has a value
        if (order.price) {
          grpcOrder.setPrice(order.price);
        }
        grpcOrder.setSide(order.isBuy ? xudrpc.OrderSide.BUY : xudrpc.OrderSide.SELL);
        return grpcOrder;
      };
```

So you can use `const grpcMakerOrder = getGrpcOrderFromOrderInstance(makerOrder!);` and `grpcTrade.setMakerOrder(grpcMakerOrder);` - and eventually do the same for the taker order as well.